cdci_osa_plugin/osa_dataserve_dispatcher.py

```diff
@@ -646,3 +649,32 @@ def get_col_data(t, n, default):
 
         return cat
 
+
+class OsaQuery(ProductQuery): 
+    """
+    base class for all osa
+    """
+    
+    def check_query_roles(self, provided_roles: List[str], par_dic: dict):
+        # TODO: there is this get_par_by_name there, to check if it can be made work
+        
+        max_pointings = int(par_dic.get('max_pointings', 50))
+        scw_list = par_dic['scw_list'] 
+        integral_data_rights = par_dic['integral_data_rights'] 
```

Comment:
Role are still necessary and indeed the only way to define privileges. 
See how required roles are compared with what is provided from the token (passed in "provided_roles").
See how the test which I added does not currently provide roles in token, which results in some parameters being correctly rejected - since the requests in this test are public, token-less.
Test with token should be also introduced.

Consider that `integral_data_rights` is not a role, and not a substitute for the role. It is a parameter. Parameters are requested by the user, and supported by the token.
In principle, it would be possible to deduce request details from the token roles, e.g. if token does not allow private data we deduce that this is what is requested. But I think this would be too confusing and lead to errors.
Instead, user can explicitly request private data in the parameters, be conscious and responsible for the request.
Also, privileged user may want to request public-only data - quite common goal.
So, I think we should be aware of difference between request parameter requesting private data and role encoded in the token which allows.